web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -838,4 +841,69 @@ public ResponseEntity<List<GenericAssayDataBin>> fetchGenericAssayDataBinCounts(
+    @RequestMapping(value = "/clinical-data-table/fetch", method = RequestMethod.POST,
+        consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Fetch clinical data for the Clinical Tab of Study View")
+    public ResponseEntity<ClinicalDataCollection> fetchClinicalDataClinicalTable(
+        @ApiParam(required = true, value = "Study view filter")
+        @Valid @RequestBody(required = false) StudyViewFilter studyViewFilter,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
+        @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
+        @Valid @RequestAttribute(required = false, value = "interceptedStudyViewFilter") StudyViewFilter interceptedStudyViewFilter,
+        @ApiParam("Page size of the result list")
+        @Max(CLINICAL_TAB_MAX_PAGE_SIZE)
+        @Min(PagingConstants.NO_PAGING_PAGE_SIZE)
+        @RequestParam(defaultValue = PagingConstants.DEFAULT_NO_PAGING_PAGE_SIZE) Integer pageSize,
+        @ApiParam("Page number of the result list")
+        @Min(PagingConstants.MIN_PAGE_NUMBER)
+        @RequestParam(defaultValue = PagingConstants.DEFAULT_PAGE_NUMBER) Integer pageNumber,
+        @ApiParam("Search term to filter sample rows. Samples are returned " +
+            "with a partial match to the search term for any sample clinical attribute.")
+        @RequestParam(defaultValue = "") String searchTerm,
+        @ApiParam("Name of the property that the result list is sorted by")
+        @RequestParam(required = false) ClinicalDataSortBy sortBy,
+        @ApiParam("Direction of the sort")
+        @RequestParam(defaultValue = "ASC") Direction direction) {
+
+        List<String> sampleStudyIds = new ArrayList<>();
+        List<String> sampleIds = new ArrayList<>();
+        List<SampleIdentifier> filteredSampleIdentifiers = studyViewFilterApplier.apply(interceptedStudyViewFilter);
+        studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, sampleStudyIds, sampleIds);
+        
+        List<ClinicalData> sampleClinicalData = clinicalDataService.fetchSampleClinicalDataClinicalTable(
+            sampleStudyIds,
+            sampleIds,
+            pageSize,
+            pageNumber,
+            searchTerm,
+            sortBy != null ? sortBy.getOriginalValue() : null,
+            direction.name());
+            
+        // Return empty when possible.
+        if (sampleClinicalData.isEmpty()) {
+            return new ResponseEntity<>(new ClinicalDataCollection(), HttpStatus.OK);
+        }
+
+        // Resolve for which patient clinical data should be included.
+        final List<ImmutablePair<String, String>> patientIdentifiers = sampleClinicalData.stream()
```

Comment:
Why are you explicitly declaring function scoped, effectively final variables as final? Not really a problem, just curious.